### PR TITLE
Update wgpu-hal to 0.16.1 to fix mobile Safari

### PR DESCRIPTION
**Pull request by emilk**
The release contains https://github.com/gfx-rs/wgpu/pull/3780

Closes https://github.com/rerun-io/rerun/issues/2032

### Checklist
* [x] I have read and agree to [Contributor Guide](https://github.com/rerun-io/rerun/blob/main/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/rerun-io/rerun/blob/main/CODE_OF_CONDUCT.md)

PR Build Summary: https://build.rerun.io/pr/2296

Docs preview: https://rerun.io/preview/9ebf96a/docs

### CODE REVIEW

File: crates/re_build_web_viewer/src/lib.rs [hunk 1]
```diff
@@ -108,13 +108,27 @@ pub fn build(release: bool, webgpu: bool) -> anyhow::Result<()> {
...
+        if err
+            .to_string()
+            .starts_with("cannot import from modules (`env`")
+        {
+            // Very common error: "cannot import from modules (`env`) with `--no-modules`"
+            anyhow::bail!(
+                "Failed to run wasm-bindgen: {err}. This is often because some dependency is calling `std::time::Instant::now()` or similar. You can try diagnosing this with:\n\
+                wasm2wat {target_wasm_path} | rg '\"env\"'\n\
+                wasm2wat {target_wasm_path} | rg 'call .now\\b' -B 20"
+            );
```
Comment (Wumpf) [inline crates/re_build_web_viewer/src/lib.rs:127]: very nice, didn't know about wasm2wat (I use a VSCode plugin for that sort of thing so far)

**Review by Wumpf — APPROVED** [review]
👍

File: crates/re_build_web_viewer/src/lib.rs (hunk 1 above)
Comment (emilk) [reply crates/re_build_web_viewer/src/lib.rs:127]: What plugin?

Comment (Wumpf) [reply crates/re_build_web_viewer/src/lib.rs:127]: "the" webassembly extension can convert wasm files https://marketplace.visualstudio.com/items?itemName=dtsvet.vscode-wasm

**Pull request merged**